python/ray/data/context.py

```diff
@@ -65,6 +65,9 @@
 # Whether to use Polars for tabular dataset sorts, groupbys, and aggregations.
 DEFAULT_USE_POLARS = False
 
+# Whether to do sampling to estimate in-memory data size for data source.
+DEFAULT_SAMPLING_DATA_SOURCE_ENABLED = True
```

Comment:
```suggestion
DEFAULT_DECODING_SIZE_ESTIMATION_ENABLED = True
```

How about we call this decoding size estimation? Sampling is just a implementation technique.